src/xtgeo/grid3d/_gridprop_lowlevel.py

```diff
@@ -76,13 +96,13 @@ def update_carray(self, undef=None, discrete=None, dtype=None, order="F"):
         dstatus = bool(discrete)
 
     if undef is None:
-        undef = xtgeo.UNDEF
+        undef = UNDEF
         if dstatus:
-            undef = xtgeo.UNDEF_INT
+            undef = UNDEF_INT
 
     logger.debug("Entering conversion from numpy to C array ...")
 
-    values = self._values.copy()
+    values = self._values.copy()  # type: ignore[has-type]
```

Comment:
I don't know the specifics of this ignore but maybe `self.values.copy()` from the getter here might help?